internal/ingress/controller/template/template.go

```diff
@@ -317,16 +317,23 @@ func locationConfigForLua(l interface{}, a interface{}) string {
 		return "{}"
 	}
 
+	pathType := ""
+	if location.PathType != nil {
+		pathType = fmt.Sprintf("%v", *location.PathType)
+	}
+
```

Comment:
Add
```
	if needsRewrite(location) || location.Rewrite.UseRegex {
		pathType = ""
	}
```

to not set Prefix as value

Reply:
would it not make more sense to have that logic in controller.go when we actually set `location.PathType`?

Reply:
We only handle Prefix and Exact. There is no case for "regex" or "rewrite"

Reply:
Semantically speaking should regex case not be handled by `pathType` `ImplementationSpecific`? In other words `use-regex` annotation would be valid only when `pathType` is set to `ImplementationSpecific`.

Reply:
No, unless we only support k8s v1.19 where the ingress path definition validation is removed
(the regex definition validation runs in go, which is not equal to the one provided by nginx)